### PR TITLE
previewableOperation API to abstract makeSavePoint

### PR DESCRIPTION
**Pull request by fdardenne**

### CODE REVIEW

File: addons/html_editor/static/src/core/history_plugin.js [hunk 1]
```diff
@@ -759,6 +765,33 @@ export class HistoryPlugin extends Plugin {
...
+     * @returns {PreviewableOperation}
+     */
+    makePreviewableOperation(operation) {
+        let revertOperation = () => {};
+        const preview = (...args) => {
+            revertOperation();
+            revertOperation = this.makeSavePoint();
+            operation(...args);
+        };
+        const apply = (...args) => {
```
Comment (ged-odoo) [inline addons/html_editor/static/src/core/history_plugin.js]: avoid using apply

Comment (ged-odoo) [reply addons/html_editor/static/src/core/history_plugin.js]:
```js
return {
  preview(...args) {
  },
};
```

Comment (ged-odoo) [reply addons/html_editor/static/src/core/history_plugin.js]: apply => commit

File: addons/html_editor/static/src/core/history_plugin.js [hunk 2]
```diff
@@ -759,6 +765,30 @@ export class HistoryPlugin extends Plugin {
...
+                revertOperation();
+                revertOperation = this.makeSavePoint();
+                operation(...args);
+            },
+            commit: (...args) => {
+                revertOperation();
+                operation(...args);
+                this.addStep();
+            },
+            revert: (...args) => {
```
Comment (Goaman) [inline addons/html_editor/static/src/core/history_plugin.js]: why ...args here?

**Pull request merged**